packages/graphql-elasticsearch-transformer/src/resources.ts

```diff
@@ -348,54 +334,37 @@ export class ResourceFactory {
     }
 
     /**
-     * Create the elasticsearch domain.
+     * If there is an env, allow ES to create the domain name so we don't go
+     * over 28 characters. If there is no env, fallback to original behavior.
      */
-    public makeElasticSearchDomain() {
-        return new Elasticsearch.Domain({
-            DomainName: Fn.Join(
-                "-",
+    private domainName() {
+        return Fn.If(
+            ResourceConstants.CONDITIONS.HasEnvironmentParameter,
+            Refs.NoValue,
+            Fn.Join(
+                '-',
                 [
-                    "d",
+                    'd',
```

Comment:
What is this for?

Reply:
The api id is 26 characters so this was just to give the cluster a slightly different name than api id.